### ISSUE TITLE
H: Stochastic gradient descent

### ISSUE THREAD

**Issue by SigurdRiseth** [SigurdRiseth/FYS-STK4155-P1#11]
Our last gradient step is to include stochastic gradient descent (Section 4.7 of the lecture notes) using the same methods to update the learning rates as in parts e)--g).  Split the training data into mini-batches, loop over epochs, and study the dependence on the mini-batch size, the number of epochs and the learning-rate schedule.  Compare and discuss your results with and without stochastic gradient descent and give a critical assessment of the various methods -- both in terms of the final accuracy relative to the closed-form solutions of parts a) and b), and in terms of computational cost.
